### PR TITLE
[cker] Make Mul support bool type

### PR DESCRIPTION
**Pull request by ragmani**
This commit makes the kernel for Mul support boolean type too.
  - Add mul op unit tests
  - Support bool type

ONE-DCO-1.0-Signed-off-by: ragmani <ragmani0216@gmail.com>

### CODE REVIEW

File: compute/cker/src/Mul.test.cc
Comment (ragmani) [inline compute/cker/src/Mul.test.cc:2]:
```suggestion
 * Copyright (c) 2023 Samsung Electronics Co., Ltd. All Rights Reserved
```

**Comment by glistening** [comment]
@ragmani At first, I thought you are introducing a new data type of `Mul bool`. Your commit message body seems to describe better. Could you please update commit title like _"[cker] Make Mul support bool type"_ ?

**Comment by ragmani** [comment]
> At first, I thought you are introducing a new data type of Mul bool. Your commit message body seems to describe better. Could you please update commit title like "[cker] Make Mul support bool type" ?

Changed it.

**Review by jyoungyun — APPROVED** [review]
LGTM

**Comment by glistening** [comment]
@ragmani 

Mul for bool type seems to be same to `LogicalAnd`. If so, why don't you use the existing `LogicalAnd.h`?

`LogicalAnd` does not require function call for each elementwise `and` operation while this PR requires lambda.

### This PR
```cc
return [](const bool &a, const bool &b) -> bool { return a && b; };
```

### LogicalAnd
```cc
template <typename T>
inline void LogicalAndElementwise(const Shape &shape, const T *input1_data, const T *input2_data,
                                  T *output_data)
{

  int num_elements = shape.FlatSize();

  for (int t = 0; t < num_elements; t++)
  {
    output_data[t] = input1_data[t] && input2_data[t];
  }
}
```

**Comment by ragmani** [comment]
> Mul for bool type seems to be same to LogicalAnd. If so, why don't you use the existing LogicalAnd.h?

> LogicalAnd does not require function call for each elementwise and operation while this PR requires lambda.

I had no idea about existence of `LogocalAnd` kernel.
Most `Mul` ops use `BinaryArithmetic` kernels that can process `BinaryArithmeticOpParam`, but `LogicalAnd` does not. I think it would be better to use a `BinaryArithmetic` kernel for consistency and maintenance even if the `BinaryArithmetic` kernel does not process `BinaryArithmeticOpParam` now.

**Comment by glistening** [comment]
@ragmani 
I have different view. `cker` is our abstraction of kernel computation. `cker` is there for code reuse. If same computation is required, there is no reason to have duplicated `cker` code. More code requires more maintenance cost. In this case, you added test case for bool type Mul. 

Furthermore, it is likely to add performance penalty as I mentioned above. (We don't need lambda for logical and.)

**Comment by ragmani** [comment]
@glistening 
If other members are agreed with you, I will change `Mul` bool op to use `LogicalAnd` instead of `BinaryArithmetic`.

@Samsung/one_onert Please give me your opinions.

**Comment by hseok-oh** [comment]
>Please give me your opinions.

No matter which one is selected.

**Comment by glistening** [comment]
I am waiting for 1 additional approval.

**Review by hseok-oh — APPROVED** [review]

**Pull request merged**
